### PR TITLE
Remove references to Weak/SoftCleanable from CleanerShutdown

### PR DESCRIPTION
**Pull request by pshipton**
WeakCleanable and SoftCleanable are removed in JDK16. The cleanup code
for these never worked properly in older versions, obviously it's not
needed.

Fixes https://github.com/eclipse/openj9/issues/10881

Signed-off-by: Peter Shipton <Peter_Shipton@ca.ibm.com>

### CODE REVIEW

**Comment by pshipton** [comment]
jenkins test sanity win,xlinux jdknext depends ibmruntimes/openj9-openjdk-jdk#openj9-staging

File: jcl/src/java.base/share/classes/jdk/internal/ref/CleanerShutdown.java
Comment (keithc-ca) [inline jcl/src/java.base/share/classes/jdk/internal/ref/CleanerShutdown.java:70]: I think this should use `weakRemove` rather than `phantomRemove`.

File: jcl/src/java.base/share/classes/jdk/internal/ref/CleanerShutdown.java
Comment (keithc-ca) [inline jcl/src/java.base/share/classes/jdk/internal/ref/CleanerShutdown.java:74]: I think this should use `softRemove` rather than `phantomRemove`.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: jcl/src/java.base/share/classes/jdk/internal/ref/CleanerShutdown.java
Comment (pshipton) [reply jcl/src/java.base/share/classes/jdk/internal/ref/CleanerShutdown.java:70]: I think we should just delete the weak and soft code for all versions considering it has never worked properly. Thoughts?

**Comment by pshipton** [comment]
The code is only activated by setting the system property `ibm.java9.forceCommonCleanerShutdown`.

This is set by 
cmdLineTester_GCRegressionTests - sanity
memleaksTest - sanity
thrstatetest - sanity
VmArgumentTests - extended
vmLifecyleTests - extended
TestInvtest - sanity, z/OS only

**Comment by pshipton** [comment]
jenkins test sanity,extended win,xlinux jdk8

**Comment by pshipton** [comment]
Note `ibm.java9.forceCommonCleanerShutdown` is not a documented option.

**Comment by pshipton** [comment]
jenkins test sanity,extended win,xlinux jdk11

File: jcl/src/java.base/share/classes/jdk/internal/ref/CleanerShutdown.java
Comment (keithc-ca) [reply jcl/src/java.base/share/classes/jdk/internal/ref/CleanerShutdown.java]: That seems reasonable: the broken code (if it ever executed) would throw `IllegalArgumentException`.

**Comment by pshipton** [comment]
@keithc-ca most of the testing has passed. I had to restart the windows sanity jdk11 due to a machine problem. Perhaps you are willing to merge this now without waiting for this test, or merge it in a few hours when the test is done, in order to unblock jdknext builds tonight.
https://ci.eclipse.org/openj9/job/Test_openjdk11_j9_sanity.functional_x86-64_windows_Personal/556/

**Comment by keithc-ca** [comment]
I thought the remaining testing would complete shortly; I didn't recognize the content of the log as a machine problem.
Sanity passed on xlinux: I don't expect a different result on Windows.
I'm satisfied. Merging.

**Review by keithc-ca — APPROVED** [review]

**Comment by pshipton** [comment]
For the record, the windows sanity jdk11 test passed.

**Comment by keithc-ca** [comment]
I just noticed [that](https://ci.eclipse.org/openj9/job/Test_openjdk11_j9_sanity.functional_x86-64_windows_Personal/556/).

**Pull request merged**
